src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -664,6 +667,31 @@ impl ThinPool {
         Ok(new_state)
     }
 
+    pub fn fstrim(&mut self, pool_name: &Name) -> StratisResult<()> {
```

Comment:
again, because fstrim can fail not because anything is wrong but just because the mount volume was unmounted at the wrong time. If writing the mdv fails then it's because something went wrong, not just timing.